js/controller.js

```diff
@@ -202,7 +202,7 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         this.state.configLoaded = true;
         this.renderSkin();
 
-        var fullClass = (this.state.config.adScreen.showControlBar ? "" : " full");
+        var fullClass = (this.state.config.adScreen.showControlBar && this.state.showAdControls ? "" : " full");
```

Comment:
Without knowing the timing of when the SHOW_AD_CONTROLS message comes through, I assumed this was the safer way to do it. Will take another look.